From: Hannes Reinecke <hare@suse.de>
Date: Thu, 10 Jan 2013 14:50:33 +0100
Subject: dasd: Disable block timeouts per default
Patch-Mainline: n/a
References: FATE#311379

block timeouts should be disabled per default; needs to be
enabled via sysfs.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/s390/block/dasd.c |   13 +++++++------
 1 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/s390/block/dasd.c b/drivers/s390/block/dasd.c
index 9d12008..a6f4ebc 100644
--- a/drivers/s390/block/dasd.c
+++ b/drivers/s390/block/dasd.c
@@ -2293,6 +2293,8 @@ enum blk_eh_timer_return dasd_times_out(struct request *req)
 		return BLK_EH_RESET_TIMER;
 
 	device = cqr->startdev ? cqr->startdev : block->base;
+	if (!device->blk_timeout)
+		return BLK_EH_RESET_TIMER;
 	DBF_DEV_EVENT(DBF_WARNING, device,
 		      " dasd_times_out cqr %p status %x",
 		      cqr, cqr->status);
@@ -2398,14 +2400,13 @@ static void dasd_setup_queue(struct dasd_block *block)
 	blk_queue_segment_boundary(block->request_queue, PAGE_SIZE - 1);
 	blk_queue_rq_timed_out(block->request_queue, dasd_times_out);
 	/*
-	 * We always set the failfast timeout;
-	 * dasd_times_out() will ignore requests when failfast
-	 * is not enabled.
+	 * blk_timeout must be enabled via sysfs; set defaults to
+	 * expires * (retries + 1)
 	 */
-	block->base->blk_timeout = block->base->failfast_expires *
-	    (block->base->failfast_retries + 1);
 	blk_queue_rq_timeout(block->request_queue,
-			     block->base->blk_timeout * HZ);
+			     block->base->default_expires *
+			     (block->base->default_retries + 1));
+	block->base->blk_timeout = 0;
 }
 
 /*
-- 
1.7.4.2

